corehq/motech/fhir/tasks.py

```diff
@@ -394,5 +403,273 @@ def create_parent_indices(
     )
 
 
+def handle_response(response):
+    if response.status_code == 200:
+        return response.json()
+    elif response.status_code >= 400:
+        response.raise_for_status()
+
+
+def generate_epic_jwt():
+    key = settings.EPIC_PRIVATE_KEY
+    # token will expire in 4 mins
+    exp = int(time.time()) + 240
+    jti = str(uuid.uuid4())
+    header = {
+        "alg": "RS256",
+        "typ": "JWT",
+    }
+    payload = {
+        "iss": settings.EPIC_CLIENT_ID,
+        "sub": settings.EPIC_CLIENT_ID,
+        "aud": "https://fhir.epic.com/interconnect-fhir-oauth/oauth2/token",
+        "jti": jti,
+        "exp": exp
+    }
+    token = jwt.encode(payload, key, algorithm="RS256", headers=header)
+    return token
+
+
+def request_epic_access_token():
+    headers = {
+        "Content_Type": "application/x-www-form-urlencoded",
+    }
+    data = {
+        "grant_type": "client_credentials",
+        "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+        "client_assertion": generate_epic_jwt()
+    }
+    url = "https://fhir.epic.com/interconnect-fhir-oauth/oauth2/token"
+    response = requests.post(url, data=data, headers=headers)
+    response_json = handle_response(response)
+
+    return response_json.get('access_token')
+
+
+def get_patient_fhir_id(given_name, family_name, birthdate, access_token):
+    base_url = "https://fhir.epic.com/interconnect-fhir-oauth/api/FHIR/R4/Patient"
+    params = {
+        'birthdate': birthdate,
+        'family': family_name,
+        'given': given_name,
+        '_format': 'json'
+    }
+    url = f'{base_url}?{urlencode(params)}'
+    headers = {
+        'authorization': f'Bearer {access_token}',
+    }
+    response = requests.get(url, headers=headers)
+    response_json = handle_response(response)
+    fhir_id = None
+    entry_list = response_json.get('entry')
+    if entry_list and len(entry_list) > 0:
+        entry = entry_list[0]
+    else:
+        entry = None
+    if entry:
+        resource = entry.get('resource')
+        if resource:
+            fhir_id = resource.get('id')
+    return fhir_id
+
+
+# TODO add time param 12 weeks from study start date
+def get_epic_appointments_for_patient(fhir_id, access_token):
+    appointments = []
+    headers = {
+        'authorization': f'Bearer {access_token}',
+    }
+    base_url = "https://fhir.epic.com/interconnect-fhir-oauth/api/FHIR/R4/Appointment"
+    params = {
+        'patient': fhir_id,
+        'service-category': 'appointment',
+        '_format': 'json'
+    }
+    url = f'{base_url}?{urlencode(params)}'
+    response = requests.get(url, headers=headers)
```

Comment:
```suggestion
    response = requests.get(base_url, params=params, headers=headers)
```

... but again, don't use `requests` directly, use the `Requests` class to get built-in authentication, and logging that you can audit later.